core/src/main/scala/iliad/Bounded.scala

```diff
@@ -7,3 +7,12 @@ trait Bounded[A] {
   def MinValue: A
   def MaxValue: A
 }
+
+object Bounded {
+def index[A](n: Int)(implicit B: Bounded[A], E: Enum[A]) = {
+    @annotation.tailrec
+    def go(a: A, n: Int): A = if(n <= 0) a else go(Enum[A].succ(a), n - 1)
+    go(Bounded[A].MinValue, n)
+
+  }
+}
```

Comment:
Can change to `A : Bounded : Enum`. It's good where it is, though would be worth returning an `Option` if it goes above max value.